pkg/project/resource_project.go

```diff
@@ -217,9 +217,22 @@ func projectResource() *schema.Resource {
 		},
 
 		"repos": {
-			Type:        schema.TypeSet,
-			Optional:    true,
-			Elem:        &schema.Schema{Type: schema.TypeString},
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+			MinItems: 0,
+			MaxItems: func() int {
+				defaultMaxItems := 100
+				return func() int {
+					value, err := getEnvIntVar("REPO_LIMIT_OVERRIDE", defaultMaxItems)
```

Comment:
This should be a bool or the user experience will be terrible.

Consider: They add 383 repos to the `repos` attribute - except, it's dynamic and he doesn't actually know how many there are. In this scenario, he would have to guess and set the env var, or more likely, he would just set a ridiculously high number - effectively treating it as a boolean.

Reply:
Initial requirement was not clear to me. Now clear.